### DISCUSSION TITLE
How functional is this provider with PVE 8.2?

### DISCUSSION THREAD

**Discussion by n8rade** [Telmate/terraform-provider-proxmox#1085] [General]
I've been pushing off my upgrade to 8.X because of some issues I've been seeing here and there, but I'm really just not sure how functional this might be with the latest version. Is anyone using a 3.0.1 release candidate with 8.X and running everything mostly okay? I can understand running into a few issues here and there, but I use my homelab to help with my work so have a fairly high desire for something functional.

I'm currently using 3.0.1-rc1 with PVE 7.4-18 with no issues and I think there's only a few more changes in rc3 I'd have to make. I'm a heavy cloud-init user as well if that matters.

Is it worth waiting for a 3.0.1 proper release before upgrading?

First time using these discussions as well so if there's a better place to ask this please point me in that direction!

**Comment by Tinyblargon** [comment]
Hi,

Sadly, the best we can currently offer is the 3.0.0 release candidate. Patching 2.x to be compatible with proxmox 8.2 is a lot of work as the library we use to connect to proxmox has changed a lot. That bieng said the release candidate should be more stable than the old 2.x realease. If stability is of the utmost importance, you could look into setting up a virtual PVE 8.2 and testing your deployments that way.

Version 3 will be released when more bugs that require breaking changes have been resolved. If we were to release 3 today, the next release would have to be 4. We want to create a stable functional release that we can start adding features to.

**Reply by n8rade** [reply]
Apologies for any confusion, I'm currently using 3.0.1-rc1 with 7.4-18 and was just wanting to know how 3.0.1-rcX was working with 8.X. If it's generally working fine then I will proceed with upgrading PVE but I'm just trying to get a general idea of where things are at.

Stability is important but there's definitely a line that every person has to draw. I can accept some bugs but would hate to find that it just doesn't work at all for me.

If the answer is to just test I can figure something out. Just wanted to see if there were any out there running it that could share their thoughts and save me some time!

**Reply by n8rade** [reply]
I was able to test 3.01-rc3 with PVE 8.2 and ran into no issues aside from those relating to breaking changes between the release candidates.  

My apologies if my discussion seemed out of left field or that I was being overly cautious. I really wanted to ensure that the functionality I needed was working and wasn't necessarily trying to separate my homelab in multiple environments. Luckily I had a lesser-used node I upgraded first and pointed a test workspace to ensure everything worked.

I appreciate your contributions to this project and I'm looking forward to the full version 3 release! :)
